src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/OpenSslX509Encoder.cs

```diff
@@ -14,13 +14,29 @@ namespace Internal.Cryptography.Pal
 {
     internal sealed class OpenSslX509Encoder : ManagedX509ExtensionProcessor, IX509Pal
     {
-        public AsymmetricAlgorithm DecodePublicKey(Oid oid, byte[] encodedKeyValue, byte[] encodedParameters, ICertificatePal? certificatePal)
+        public ECDsa DecodeECDsaPublicKey(ICertificatePal? certificatePal)
+        {
+            if (certificatePal is OpenSslX509CertificateReader reader)
```

Comment:
Previously this was just a nullcheck then a hard cast.  Since the PAL type is internal it's... unlikely... that we'd have a second implementation in the same environment, but I wonder if the NotSupportedException if it happened would be harder to diagnose than the hard cast failing.

Reply:
Ah, the Apple one did it as `as Foo` and checked the result so I just had it stuck one way in my head. I think it makes sense to change both.

https://github.com/dotnet/runtime/blob/d0b699fd5d24f340b7a0ab4bf656f85ea9af822e/src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.OSX/X509Pal.cs#L31-L33